service_container/factories.rst

```diff
@@ -293,5 +293,138 @@ previous examples takes the ``templating`` service as an argument:
                 ->args([ref('templating')])
             ;
         };
+        
+Usage example
+-------------
+
+The following example is intended to show how to create and use a factory method in Symfony framework.
+Suppose you want to realize the factory method pattern for services, that describe two delivery methods - DHL and UPS.
+
+Services (subclasses) definition::
+
+    // src/Deliveries/DHL.php
+    namespace App\Deliveries;
+
+    class DHL
```

Comment:
To be more real life the should implement a marker interface.

Reply:
Thanks, done.